lib/pliny/librato/metrics/backend.rb

```diff
@@ -31,14 +35,15 @@ def start
 
         def stop
           metrics_queue.push(POISON_PILL)
-          timer.terminate
+          # Ensure timer is not running when we terminate it
+          sync { timer.terminate }
```

Comment:
@appleton pointed out this won't work in a signal trap. He also [suggested an alternate approach on the previous change](https://github.com/heroku/pliny-librato/pull/9#issuecomment-272398649), which will work, but it'll take up to `interval` seconds as `sleep interval` will not spuriously wake up.